op-bindings/etherscan/client.go

```diff
@@ -0,0 +1,196 @@
+package etherscan
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-service/retry"
+)
+
+type client struct {
+	baseUrl string
+}
+
+type apiResponse struct {
+	Status  string      `json:"status"`
+	Message string      `json:"message"`
+	Result  interface{} `json:"result"`
+}
+
+type rpcResponse struct {
+	JsonRpc string      `json:"jsonrpc"`
+	Id      int         `json:"id"`
+	Result  interface{} `json:"result"`
+}
+
+type TxInfo struct {
+	To    string `json:"to"`
+	Input string `json:"input"`
+}
+
+const apiMaxRetries = 3
+const apiRetryDelay = time.Duration(2) * time.Second
+const errRateLimited = "Max rate limit reached"
```

Comment:
```suggestion
const ErrRateLimited = errors.New("Max rate limit reached")
```

this way you can just return the error directly. Also should make public so that the caller can switch against the error if desired